File: server/src/main/java/org/dan/jadalnia/app/user/UserSession.java
package org.dan.jadalnia.app.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.dan.jadalnia.sys.error.TemplateError;

import static java.util.Collections.singletonMap;
import static org.dan.jadalnia.sys.error.JadEx.badRequest;

@Getter
@Setter
@Builder
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
public class UserSession {
    private Uid uid;
    private String key;

    public static UserSession valueOf(String s) {
        final String[] parts = s.split(":", 2);
        try {
            return UserSession.builder()
                    .uid(Uid.valueOf(parts[0]))
                    .key(parts[1])
                    .build();
        } catch (NumberFormatException e) {
            throw badRequest(
                    new TemplateError("user session has wrong format: [$session]",
                            singletonMap("session", s)), e);
        }
    }

    public String toString() {
        return uid + ":" + key;
    }
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/user/UserSession.kt
package org.dan.jadalnia.app.user;

import org.dan.jadalnia.sys.error.JadEx.Companion.badRequest
import org.dan.jadalnia.sys.error.TemplateError
import java.util.Collections.singletonMap

data class UserSession(val uid: Uid, val key: String) {
    companion object {
        fun valueOf(s: String): UserSession {
            val parts = s.split(Regex(":"), 2);
            try {
                return UserSession(
                        uid = Uid.valueOf(parts[0]),
                        key = parts[1])
            } catch (e: NumberFormatException) {
                throw badRequest(
                        TemplateError("user session has wrong format: [\$session]",
                                singletonMap("session", s)), e);
            }
        }
    }

    override fun toString(): String = uid + ":" + key
}
